utils/ansible/ftest/templates/daos-make.sh.j2

```diff
@@ -205,12 +229,19 @@ case $CMD in
 esac
 
 info "Building and installing DAOS from source tree $PWD to $DAOS_INSTALL_DIR"
-if ! run env MPI_PKG=any "$SCONS_EXE" BUILD_TYPE=debug PREFIX="$DAOS_INSTALL_DIR" install $SCONS_OPTS ; then
+source "$DAOS_INSTALL_DIR/bin/activate"
+if ! run env MPI_PKG=any "$SCONS_EXE" BUILD_TYPE="$BUILD_TYPE" PREFIX="$DAOS_INSTALL_DIR" install $SCONS_OPTS ; then
```

Comment:
`MPI_PKG`, 'BUILD_TYPE` and 'PREFIX' will should all be set on the build line above on L227, otherwise the tree will be built twice I'd expect?

If this is expected to do both the build and install then what is L227 doing?

Reply:
The L226 is used  for building and install a fresh DAOS and line 227 just to update a previous local install done with the line 226.  However, it is highly probable that it is the not the way to use scons as I am no used to it.

Reply:
For the general case and indeed this work there are three stages of the build that scons needs to perform

1. Building and installing all the deps.
2. Building daos
3. installing daos.

For 1. you can use `--build-deps=yes` to perform it at the same time as 2. however using `--build-deps=only` will perform 1. but not 2.
2. Options like `BUILD_TYPE`, `PREFIX` etc can be passed to any invocation and they'll be saved in `daos.conf` and used for future invocations.
For 3. if daos needs to be rebuilt because of options then it'll always rebuilt it.  As far as I'm aware there's no "install but fail if rebuild is required" option.

I think this code will call `scons --build-deps=yes` which will then build and install the dependencies, and then build daos.  It's then calling `scons` which will be a no-op as the code is built and then it's calling `scons install PREFIX=` which will re-build and then install the whole tree.

For correctness and to get any errors reported where you'd expect them and for the timings to be as expected then this code should be doing

- `scons --build-deps=only`
- `scons PREFIX=<> BUILD_TYPE=<>`
- `scons install`

This way the three commands that this script launches will match with the three build steps I illustrated above.  Rebuilds from the same source tree can miss the first step.  In theory they could skip the second as well unless the options have changed but they would want to do steps 2 and 3.

Reply:
- [ ] Properly call scons

Reply:
> * [x]  Properly call scons

Fixed with commit 8a62b1f000ca6c212e790b7aaddf992c98651969